django/views/generic/detail.py

```diff
@@ -64,13 +78,13 @@ def get_queryset(self):
         `get_object` and may not be called if `get_object` is overridden.
         """
         if self.queryset is None:
-            if self.model:
-                return self.model._default_manager.all()
+            if self.get_model():
+                return self.get_model()._default_manager.all()
             else:
                 raise ImproperlyConfigured(
                     "%(cls)s is missing a QuerySet. Define "
                     "%(cls)s.model, %(cls)s.queryset, or override "
-                    "%(cls)s.get_queryset()." % {
+                    "%(cls)s.get_queryset() or %(cls)s.get_model()." % {
                         'cls': self.__class__.__name__
                     }
                 )
```

Comment:
You'll need to catch `ImproperlyConfigured` errors raised by `get_model()` if you want the mention about `model` and `get_model()` to be meaningful.

```python
try:
    model = self.get_model()
except ImproperlyConfigured:
    model = None
if model:
    return model._default_manager.all()
else:
    raise ImproperlyConfigured(
        "..."
    )
```